app/src/main/java/com/github/sdp/tarjetakuna/ui/singlecard/SingleCardViewModel.kt

```diff
@@ -67,15 +65,27 @@ class SingleCardViewModel : ViewModel() {
      * remove it only if it's not owned
      */
     fun manageWantedCollection() {
-        val data = userDB.getCardFromCollection(FBMagicCard(card))
-        data.thenAccept {
-            val fbCard = convertJsonToFBMagicCard(it.value as String)
-            if (fbCard.possession == FBCardPossession.WANTED) {
-                removeCardFromFirebase()
+
+        viewModelScope.launch {
+            val lCard =
+                localDatabase?.magicCardDao()?.getCard(card.set.code, card.number.toString())
+            if (lCard != null) {
+                // card not wanted -> make it unwanted from the collection
+                if (lCard.possession == CardPossession.WANTED) {
+                    manageCardsInDatabase(card, CardPossession.NONE)
+                    updateButtons(CardPossession.NONE)

```

Comment:
given how many times these two are called together maybe there should be a single method that calls both?